wire-ios-sync-engine/Tests/Source/Calling/MLSConferenceStaleParticipantsRemoverTests.swift

```diff
@@ -0,0 +1,270 @@
+//
+// Wire
+// Copyright (C) 2023 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import Foundation
+import XCTest
+@testable import WireSyncEngine
+
+class MLSConferenceStaleParticipantsRemoverTests: MessagingTest {
+
+    private let domain = "example.domain.com"
+    private let groupID = MLSGroupID.random()
+    private let connectedState = CallParticipantState.connected(
+        videoState: .stopped,
+        microphoneState: .muted
+    )
+
+    private var mlsService: MockMLSService!
+    private var sut: MLSConferenceStaleParticipantsRemover!
+
+    override func setUp() {
+        super.setUp()
+        mlsService = MockMLSService()
+        sut = MLSConferenceStaleParticipantsRemover(
+            mlsService: mlsService,
+            context: uiMOC,
+            removalTimeout: 0.4
+        )
+
+        let selfUser = ZMUser.selfUser(in: uiMOC)
+        selfUser.remoteIdentifier = UUID()
```

Comment:
why do we do this?

does this need to be reverted at teardown?

Reply:
> why do we do this?

When the sut calls `excludingSelf`, it needs the self user to be set 

> does this need to be reverted at teardown?

We already set the self user in the same way in a myriad of other tests without tearing it down, so it's safe to leave it as it is here. Although ideally we probably shouldn't

Reply:
right I am sure we do it elsewhere, just wondering if it can be a cause of flakyness.

the selfUser has no remoteIdentifier by default ? does that normally come from sync ?

Reply:
I don't know